tests/services/synapse/test_ops.py

```diff
@@ -34,3 +68,155 @@ def test_monitor_evaluation_queue_returns_true_when_there_are_new_submissions(
+
+    # Mocking the table query results
+    table_mock = MagicMock()
+    table_mock.asDataFrame.return_value = pd.DataFrame(input_dict)
+
+    # Mock the ``trigger_indexing`` call in SynapseOps() and the tableQuery call
+    # in ``SynapseOps().get_submissions``
+    with mocker.patch.object(mocked_ops, "trigger_indexing", return_value=None):
+        with mocker.patch.object(syn_mock, "tableQuery", return_value=table_mock):
+            # Calling the function to be tested
+            result = mocked_ops.get_submissions(submission_view, submission_status)
+
+            # Assertions
+            syn_mock.tableQuery.assert_called_once_with(
+                f"select * from {submission_view} where status = '{submission_status}'"
+            )
+            table_mock.asDataFrame.assert_called()
+            assert result == input_dict["id"]
+
+
+def test_update_submissions_status_with_input_list(
+    mocker: pytest.fixture, mocked_ops: MagicMock
+) -> None:
+    """
+    Tests that the ``update_submissions_status`` method in ``SynapseOps``
+    updates the status of one or more submissions in Synapse.
+
+    Arguments:
+        mocker: A mocker object.
+        mocked_ops: A mocked instance of ``SynapseOps``.
+    """
+    # Mocking the ``update_submission_status`` call in ``SynapseOps``
+    mock_update_status = mocker.patch.object(mocked_ops, "update_submissions_status")
+
+    # Calling the function to be tested with an input list of strings
+    mocked_ops.update_submissions_status(["submission_1", "submission_2"], "SCORED")
+
+    # Assertions
+    mock_update_status.assert_called_once_with(
+        ["submission_1", "submission_2"], "SCORED"
+    )
+
+
+def test_update_submissions_status_with_input_string(
+    mocker: pytest.fixture, mocked_ops: MagicMock
+) -> None:
+    """
+    Tests that the ``update_submissions_status`` method in ``SynapseOps``
+    updates the status of one or more submissions in Synapse.
+
+    Arguments:
+        mocker: A mocker object.
+        mocked_ops: A mocked instance of ``SynapseOps``.
+
+    """
+    # Mocking the ``update_submission_status`` call in ``SynapseOps``
+    mock_update_status = mocker.patch.object(mocked_ops, "update_submissions_status")
+
+    # Calling the function to be tested with an input string
+    mocked_ops.update_submissions_status("submission_1", "RECEIVED")
+
+    # Assertions
+    mock_update_status.assert_called_once_with("submission_1", "RECEIVED")
+
+
+def test_update_submission_status_with_non_string_non_list_input() -> None:
+    """
+    Tests that the ``update_submissions_status`` method in ``SynapseOps``
+    raises an error if the input is neither a string nor a list.
+
+    Arguments:
+        mocker: A mocker object.
+        mocked_ops: A mocked instance of ``SynapseOps``.
+
+    """
+    # Test for non-string + non-list submission_ids.
+    error = "``submission_ids`` must be a string or list of strings."
+    with pytest.raises(TypeError) as err:
```

Comment:
I think this way of doing this is a little simpler: https://github.com/Sage-Bionetworks/agora-data-tools/blob/efe6eecf4acb0b5e1f4ff4c71ec0c59051a9c373/tests/test_utils.py#L38

Reply:
Ah, I forgot to comment on this one: So the reason I chose not to use `match` is so that I can handle the error message string-matching with my own assertion call and therefore write a more contextualized error message instead of the generic `couldn't match to Regex` assertion error.

What do you think of this? If we think the custom error message is not worth it I can use `match` instead.

Reply:
I don't really think we need to worry about this. When an assertion fails in a test it tells you the expected and actual values in the stdout